radix-engine/src/engine/process.rs

```diff
@@ -1006,11 +1012,20 @@ impl<'r, 'l, L: SubstateStore> Process<'r, 'l, L> {
 
         let data = Self::process_entry_data(&input.state)?;
         let new_objects = wasm_process.process_owned_objects.take(data)?;
+        let sys_auth: HashMap<String, AuthRule> = input
+            .sys_auth
+            .into_iter()
+            .map(|(name, auth_rule)| match auth_rule {
+                ::scrypto::resource::AuthRule::Just(addr) => {
+                    (name, AuthRule::JustNonFungible(addr))
+                }
+            })
+            .collect();
         let component = Component::new(
             wasm_process.vm.invocation.package_id,
             input.blueprint_name,
+            sys_auth,
```

Comment:
Understand the auth is enforced by system, but I'd suggest rename this one to "auth_rules".